features/authentication/README.md

```diff
@@ -1 +1,36 @@
-## Feature: Authentication
\ No newline at end of file
+# Feature: Authentication
+
+This document will reflect on how we make a full authentication flow 
+with a specified happy path. The main idea it's to establish certain
+contracts between back-end and front-end so later an example project
+can implement those contracts in their languages.
+
+## Premises
+
+There are some premises that will influence the contracts and the
+authentication flow.
+
+- Authentication after user sign in we will be done using a token, instead of cookies.
+This token should have an expiration set and it should be refresh when needed.
+- For social authentication we will use Firebase as dependency. It handles edge
```

Comment:
I will like to discuss this in depth, here are some tasks I would like to be done before making any decision:
- check that every programming language that we use has an official SDK for firebase
- check that the list of social networks we've used in the past are covered by firebase
- measure the tradeoff of using a proxy for this matter

Reply:
- Checked (All have, the only issue it's that the PHP one is unofficial)
- Checked (Attaching image with providers, I don't see one missing)
<img width="550" alt="Screenshot 2023-05-08 at 16 34 26" src="https://user-images.githubusercontent.com/13692810/236852714-573f6372-6b24-4d37-900d-c09813ed6b0e.png">
- Sure

Implementing Facebook, Apple and Google auth independently in each platform could be a big job.